workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationMiddleware.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { LoggingService } from '@aws/workbench-core-logging';
+import { Request, Response, NextFunction } from 'express';
+import { AuthenticatedUserParser } from '../models/authenticatedUser';
+import { HTTPMethodParser } from '../models/routesMap';
+import { DynamicAuthorizationService } from './dynamicAuthorizationService';
+
+/**
+ * Creates an authorization middleware function for {@link https://expressjs.com/ | express} using the {@link AuthorizationService}.
+ * @param authorizationService - {@link AuthorizationService}
+ * @returns - The authorization middleware function.
+ */
+export default function withDynamicAuth(
+  dynamicAuthorizationService: DynamicAuthorizationService,
+  options?: {
+    logger?: LoggingService;
+  }
+): (req: Request, res: Response, next: NextFunction) => Promise<void> {
+  /**
+   * Authorization Middleware
+   */
+  return async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    try {
+      const route: string = req.baseUrl + req.path;
+      const method = HTTPMethodParser.parse(req.method);
+      const { data } = await dynamicAuthorizationService.isRouteIgnored({ route, method });
+      if (data.routeIgnored) {
+        next();
+        return;
+      }
+      const authenticatedUser = AuthenticatedUserParser.parse(res.locals.user);
+      await dynamicAuthorizationService.isAuthorizedOnRoute({
+        route,
+        method,
+        authenticatedUser
+      });
+      next();
+      return;
```

Comment:
nit: dont need a return here

Reply:
Depending on how the middleware returns, it will continue execution if we don't return early

Reply:
THis is the end of the function so theres no more execution to do

Reply:
Oops, thought you were pointing to the other return haha